boards/nxp/mimxrt700_evk/board.c

```diff
@@ -0,0 +1,263 @@
+	SET_UP_FLEXCOMM_CLOCK(8);
+#endif
+
+#if DT_NODE_HAS_STATUS(DT_NODELABEL(flexcomm9), okay)
+	SET_UP_FLEXCOMM_CLOCK(9);
+#endif
+
+#if DT_NODE_HAS_STATUS(DT_NODELABEL(flexcomm10), okay)
+	SET_UP_FLEXCOMM_CLOCK(10);
+#endif
+
+#if DT_NODE_HAS_STATUS(DT_NODELABEL(flexcomm11), okay)
+	SET_UP_FLEXCOMM_CLOCK(11);
+#endif
+
+#if DT_NODE_HAS_STATUS(DT_NODELABEL(flexcomm12), okay)
+	SET_UP_FLEXCOMM_CLOCK(12);
+#endif
+
+#if DT_NODE_HAS_STATUS(DT_NODELABEL(flexcomm13), okay)
+	SET_UP_FLEXCOMM_CLOCK(13);
+#endif
+
+#if DT_NODE_HAS_STATUS(DT_NODELABEL(lpspi14), okay)
+	CLOCK_EnableClock(kCLOCK_LPSpi14);
+	RESET_ClearPeripheralReset(kLPSPI14_RST_SHIFT_RSTn);
+#endif
+
+#if DT_NODE_HAS_STATUS(DT_NODELABEL(lpi2c15), okay)
+	CLOCK_EnableClock(kCLOCK_LPI2c15);
+	RESET_ClearPeripheralReset(kLPI2C15_RST_SHIFT_RSTn);
+#endif
+
+#if DT_NODE_HAS_STATUS(DT_NODELABEL(lpspi16), okay)
+	CLOCK_AttachClk(kFRO0_DIV1_to_LPSPI16);
+	CLOCK_SetClkDiv(kCLOCK_DivLpspi16Clk, 1U);
+	CLOCK_EnableClock(kCLOCK_LPSpi16);
+	RESET_ClearPeripheralReset(kLPSPI16_RST_SHIFT_RSTn);
+#endif
+
+#if DT_NODE_HAS_STATUS(DT_NODELABEL(flexcomm17), okay)
+	CLOCK_AttachClk(kSENSE_BASE_to_FLEXCOMM17);
+	CLOCK_SetClkDiv(kCLOCK_DivLPFlexComm17Clk, 4U);
+#endif
+
+#if DT_NODE_HAS_STATUS(DT_NODELABEL(flexcomm18), okay)
+	CLOCK_AttachClk(kSENSE_BASE_to_FLEXCOMM18);
+	CLOCK_SetClkDiv(kCLOCK_DivLPFlexComm18Clk, 4U);
+#endif
+
+#if DT_NODE_HAS_STATUS(DT_NODELABEL(flexcomm19), okay)
+	CLOCK_AttachClk(kSENSE_BASE_to_FLEXCOMM19);
+	CLOCK_SetClkDiv(kCLOCK_DivLPFlexComm19Clk, 4U);
+#endif
+
+#if DT_NODE_HAS_STATUS(DT_NODELABEL(flexcomm20), okay)
+	CLOCK_AttachClk(kSENSE_BASE_to_FLEXCOMM20);
+	CLOCK_SetClkDiv(kCLOCK_DivLPFlexComm20Clk, 4U);
+#endif
+
+#if DT_NODE_HAS_STATUS(DT_NODELABEL(gpio0), okay)
+	CLOCK_EnableClock(kCLOCK_Gpio0);
+	RESET_ClearPeripheralReset(kGPIO0_RST_SHIFT_RSTn);
+#endif
+
+#if DT_NODE_HAS_STATUS(DT_NODELABEL(gpio1), okay)
+	CLOCK_EnableClock(kCLOCK_Gpio1);
+	RESET_ClearPeripheralReset(kGPIO1_RST_SHIFT_RSTn);
+#endif
+
+#if DT_NODE_HAS_STATUS(DT_NODELABEL(gpio2), okay)
+	CLOCK_EnableClock(kCLOCK_Gpio2);
+	RESET_ClearPeripheralReset(kGPIO2_RST_SHIFT_RSTn);
+#endif
+
+#if DT_NODE_HAS_STATUS(DT_NODELABEL(gpio3), okay)
+	CLOCK_EnableClock(kCLOCK_Gpio3);
+	RESET_ClearPeripheralReset(kGPIO3_RST_SHIFT_RSTn);
+#endif
+
+#if DT_NODE_HAS_STATUS(DT_NODELABEL(gpio4), okay)
+	CLOCK_EnableClock(kCLOCK_Gpio4);
+	RESET_ClearPeripheralReset(kGPIO4_RST_SHIFT_RSTn);
+#endif
+
+#if DT_NODE_HAS_STATUS(DT_NODELABEL(gpio5), okay)
+	CLOCK_EnableClock(kCLOCK_Gpio5);
+	RESET_ClearPeripheralReset(kGPIO5_RST_SHIFT_RSTn);
+#endif
+
+#if DT_NODE_HAS_STATUS(DT_NODELABEL(gpio6), okay)
+	CLOCK_EnableClock(kCLOCK_Gpio6);
+	RESET_ClearPeripheralReset(kGPIO6_RST_SHIFT_RSTn);
+#endif
+
+#if DT_NODE_HAS_STATUS(DT_NODELABEL(gpio7), okay)
+	CLOCK_EnableClock(kCLOCK_Gpio7);
+	RESET_ClearPeripheralReset(kGPIO7_RST_SHIFT_RSTn);
+#endif
+
+#if DT_NODE_HAS_STATUS(DT_NODELABEL(gpio8), okay)
```

Comment:
use if() not #if  to increase compilation coverage and reduce conditional code (compiler optimizes away things in if(0))

Reply:
I have tried it, however I meet an issue when changed to not use conditional code. Due to domain, cpu0/1 can't access all the peripheral, if I use  C if as judgement, C complier still check the element under the if statement. So it means that C compiler will report an error don't find IP cpu1 can access(however, cpu0 can't) when I compile a cpu0 image.

Reply:
This seems impossible because the premise for using this is that DT_DRV_COMPAT is unique, but there are too many types of IP in board.c.